Formula/libvisual-projectm.rb

```diff
@@ -0,0 +1,44 @@
+class LibvisualProjectm < Formula
+  desc "Visualization plug-in for projectM support from Libvisual"
+  homepage "https://github.com/projectM-visualizer/frontend-libvisual-plug-in"
+  url "https://github.com/projectM-visualizer/frontend-libvisual-plug-in/archive/refs/tags/v2.1.1.tar.gz"
+  sha256 "eb8269c2a923546600d3f40ff90c011f03a215847215ee8bef44bfae305b4dd7"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "libvisual"
+  depends_on "projectm"
+
+  def install
+    # We cannot write to libvisual's cellar, so we need to deflect
+    # installation and ask the user to create a symlink via caveats.
+    inreplace "CMakeLists.txt",
+      "LIBRARY DESTINATION \"${LIBVISUAL_ACTOR_PLUGINS_DIR}\"",
+      "LIBRARY DESTINATION \"#{lib}\""
+
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    doc.mkpath
+    cp "AUTHORS.md", doc/"AUTHORS"
+  end
+
+  def caveats
+    libvisual = Formula["libvisual"]
+    libvisual_actor_plugins_dir = "#{libvisual.lib}/libvisual-#{libvisual.version.major_minor}/actor"
+    <<~EOS
+      This formula needs your help with making libvisual find the plug-in.
+      To create the needed symlink, please execute:
+        ln -s "#{lib}/libprojectM_libvisual.so" "#{libvisual_actor_plugins_dir}/libprojectM_libvisual.so"
```

Comment:
Can we write this symlink ourselves in `post_install`?

Or, alternatively: does `libvisual` install naything itself into `libvisual.lib/"libvisual-#{libvisual.version.major_minor}/actor"`?

Reply:
Hi @carlocab , thanks for you reply!

> Can we write this symlink ourselves in `post_install`?

I tried that now and it works well locally, but CI [rejects it](https://github.com/Homebrew/homebrew-core/actions/runs/4781147382/jobs/8499397502#step:8:79) saying:
> ```
>  ==> An exception occurred within a child process:
>    Errno::EPERM: Operation not permitted @ rb_file_s_symlink - (../../../../../libvisual-projectm/2.1.1/lib/libprojectM_libvisual.so, /opt/homebrew/Cellar/libvisual/0.4.2/lib/libvisual-0.4/actor/libprojectM_libvisual.so)
> ```

I guess that means reverting from post_install to caveat?

> Or, alternatively: does `libvisual` install naything itself into `libvisual.lib/"libvisual-#{libvisual.version.major_minor}/actor"`?

With the current libvisual-plugins as a resource it does, yes. What did you have in mind?

Reply:
Yea, I guess keep the `caveat`s for now.

Reply:
@carlocab back to caveat now, done :+1: